refactor(MediaRequest): consolidate lifecycle hooks into subscriber & restrict cascade to prevent recursion

fallenbagel opened the pull request:
#### Description
This PR consolidates all database‐accessing lifecycle hooks into the `MediaRequestSubscriber` as this is the recommended way of doing database calls. 
See: https://orkhan.gitbook.io/typeorm/docs/listeners-and-subscribers#what-is-an-entity-listener.

It also tightens the cascade options on the `MediaRequest→Media` relation to only `insert` and `remove` to avoid`hook‐within‐hook` recursions that could cause unexpected behavior. 

These changes should guard against recursive hook invocations in any database backend and simplify the entity lifecycle.

Reference: https://github.com/fallenbagel/jellyseerr/pull/1619

#### Screenshot (if UI-related)

#### To-Dos

- [ ] Successful build `yarn build`
- [ ] Translation keys `yarn i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #XXXX

gauthier-th committed:
fix(mediarequest): move methods modifying MediaRequest to its Subscriber

fallenbagel committed:
fix(mediasubscriber): use event.manager for parent media updates on remove

Replace `getRepository(Media)` calls with `event.manager` in the `afterRemove` hook so that
parent-media status resets run within the same transaction/QueryRunner (important for postgresql.
Doesnt affect sqlite).

fallenbagel committed:
fix(mediarequest entity): narrow cascade to insert & remove to prevent hook recursion

Restrict cascade options on the MediaRequest→Media relation to only `insert` and `remove`to avoid
nested subscriber/AfterUpdate recursion when saving entities.

OwsleyJr commented on server/subscriber/MediaRequestSubscriber.ts:
```diff
@@ -125,6 +773,28 @@ export class MediaRequestSubscriber
     }
   }
 
+  public afterInsert(event: InsertEvent<MediaRequest>): void {
+    if (!event.entity) {
+      return;
+    }
+
+    this.sendToRadarr(event.entity as MediaRequest);
+    this.sendToSonarr(event.entity as MediaRequest);
+
+    this.updateParentStatus(event.entity as MediaRequest);
+  }
```
Do we want to make afterUpdate and afterInsert here async?

fallenbagel replied:
Oh yeah! Both `afterInsert` and `afterUpdate` should be async and await the async methods inside. Otherwise, we might get unhandled promises or race conditions 🤔

I'll update it today

fallenbagel replied:
@OwsleyJr Should be resolved in the latest commit

fallenbagel committed:
refactor(mediasubscriber): make afterInsert and afterUpdate async and await internal operations

This should prevent unhandled promise rejections and ensure sequential execution of lifecycle
actions
